src/plugins/intel_cpu/src/nodes/kernels/x64/random_uniform.cpp

```diff
@@ -626,10 +625,133 @@ void RandomUniform<isa>::tail(const std::vector<Vmm>& vmm_dst) {
     L(l_end);
 }
 
-template class RandomUniform<x64::avx512_core>;
-template class RandomUniform<x64::avx2>;
-template class RandomUniform<x64::sse41>;
+template class PhiloxGenerator<x64::avx512_core>;
+template class PhiloxGenerator<x64::avx2>;
+template class PhiloxGenerator<x64::sse41>;
+
+#undef GET_OFF
+
+//////////////// MERSENNE TWISTER GENERATOR ////////////////////
+
+#define GET_OFF(field) offsetof(MersenneTwisterGeneratorCallArgs, field)
+
+template <x64::cpu_isa_t isa>
+MersenneTwisterGenerator<isa>::MersenneTwisterGenerator(const GeneratorCompileParams& jcp) :
+        JitKernel(jit_name(), jcp, isa) {
+}
+
+template <x64::cpu_isa_t isa>
+void MersenneTwisterGenerator<isa>::generate() {
+    this->preamble();
+    registersPool = RegistersPool::create(isa, {rax, rcx, rsp, rdi, k0});
+
+    r64_dst = getReg64();
+    r64_work_amount = getReg64();
+
+    mov(r64_work_amount, ptr[r64_params + GET_OFF(work_amount)]);
+    mov(r64_dst, ptr[r64_params + GET_OFF(dst_ptr)]);
+
+    initVectors();
+    process();
+
+    registersPool.reset();
+    this->postamble();
+}
+
+template <>
+void MersenneTwisterGenerator<x64::avx512_core>::initVectors() {
+    const auto r64_aux = getReg64();
+    const auto r32_aux = Xbyak::Reg32(r64_aux.getIdx());
+
+    v_state = getVmm();
+    v_mt_index = getVmm();
+    v_temp = getVmm();
+
+    // Initialize the state array.
+    mov(r64_aux, ptr[r64_params + GET_OFF(state_ptr)]);
+    uni_vmovdqu(v_state, ptr[r64_aux]);
+
+    // Initialize constants
+    mov(r32_aux, MT_CONST_1);
+    uni_vpbroadcastd(v_temp, r32_aux);
+}
+
+template <x64::cpu_isa_t isa>
+void MersenneTwisterGenerator<isa>::initVectors() {
+    const auto r64_aux = getReg64();
+
+    v_state = getVmm();
+    v_mt_index = getVmm();
+    v_temp = getVmm();
+
+    // Initialize the state array.
+    mov(r64_aux, ptr[r64_params + GET_OFF(state_ptr)]);
+    uni_vmovdqu(v_state, ptr[r64_aux]);
+
+    // Initialize constants
+    mov(r64_aux, MT_CONST_1);
+    uni_vmovdqu(v_temp, ptr[r64_aux]);
+}
+
+template <x64::cpu_isa_t isa>
+void MersenneTwisterGenerator<isa>::process() {
+    auto v_dst_0 = getVmm();
+    auto v_dst_1 = getVmm();
+
+    const auto VECTOR_LENGTH = 10ul;
```

Comment:
What is the magic value?
Do you mean vector register length (looks so)?
on `sse4.2` the len is `128` bits - `16` bytes - 4 float values
`avx2`- `256` bits - `32` bytes - 8 float values
`avx512` - `512` bits - `64` bytes - 16 float values
`size_t vlen = cpu_isa_traits<isa>::vlen;`

Reply:
Oh, thank you for the helper function